src/BSPTree.h

```diff
@@ -75,11 +83,31 @@ class CBSPTree
 
 		// else -> prepare for creating a branch node
 		// First split the bounding volume into two halfes
-		int     splitDim = MaxDim(box.getMaxPoint() - box.getMinPoint());                   // Calculate split dimension as the dimension where the aabb is the widest
-		float   splitVal = (box.getMinPoint()[splitDim] + box.getMaxPoint()[splitDim]) / 2; // Split the aabb exactly in two halfes
-		auto    splitBoxes = box.split(splitDim, splitVal);
+		int splitDim = MaxDim(box.getMaxPoint() - box.getMinPoint());                  
+		// Calculate split dimension as the dimension where the aabb is the widest
+		//----------------------------------------------------------------------
+		std::vector<double> prim_dist;  //we store the values of primitive distances in this vector
+		for (auto pPrim : vpPrims) { //split values are going to be balanced equally
+			//we find the primitive points is the centre 
+			Vec3f prim_p = ((pPrim->getBoundingBox().getMaxPoint() - //we check how many prims we have
```

Comment:
should be plus here

Reply:
comments make no sense. 
I believe this code was copied from the solution of another student, the code was not clear and thus it contains comments, which make no sense